src/core/qgsrendercontext.cpp

```diff
@@ -381,13 +383,18 @@ double QgsRenderContext::convertToPainterUnits( double size, QgsUnitTypes::Rende
       convertedSize = std::min( convertedSize, scale.maxSizeMM * mScaleFactor );
   }
 
+  const double symbologyReferenceScaleFactor = mSymbologyReferenceScale > 1 ? mSymbologyReferenceScale / mRendererScale : 1;
```

Comment:
Shouldn't the conditional check be something like mSymbologyReferenceScale != -1 ?

Reply:
Yep, fixed